src/InverseKinematicsIPOPT.cpp

```diff
@@ -0,0 +1,499 @@
+#include "InverseKinematicsIPOPT.h"
+
+#include <cassert>
+
+using namespace std;
+using namespace iDynTree;
+using namespace Ipopt;
+
+InverseKinematicsIPOPT::InverseKinematicsIPOPT()
+: exitCode(-12)
+, parentFrame(0)
+, endEffectorFrame(0)
+,modelLoaded(false)
+,framesLoaded(false)
+,gainsLoaded(false)
+,angleError(0)
+{
+    positionError.zero();
+    rotationError.Identity();
+}
+
+InverseKinematicsIPOPT::~InverseKinematicsIPOPT()
+{}
+
+void InverseKinematicsIPOPT::removeJoints(const Model modelInput)
+{
```

Comment:
All joints which has 0 or more than 1 DOF. I did it in order to have a one to one relation between joints and dofs: with 0 DOF joints they are completely useless when computing IK, while for more than one DOF joints, there's no support from ``iDynTree``.